### PR TITLE
LinearAlgebra.BLAS: sort and group blas functions

### PR DESCRIPTION
**Pull request by inkydragon**
- [x] No additions or deletions.
- [x] stdlib: Sorted exported functions
- [x] doc: BLAS functions are grouped and sorted
- [x] Added placeholders for missing BLAS functions

group and sort ref: http://www.netlib.org/blas/blasqr.pdf

### CODE REVIEW

**Comment by ViralBShah** [comment]
Off-topic, but thinking about further refactoring, we do want to move `LinearAlgebra` into its own repo (but it will still be an stdlib and in the system image).

**Comment by inkydragon** [comment]
> we do want to move `LinearAlgebra` into its own repo

I can't wait.
Now the [`LinearAlgebra` document](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra) is very looooong, and we only show the `<h2>` headings (waiting for [right side TOC](https://github.com/JuliaDocs/Documenter.jl/issues/1688)), which makes this document hard to read.

If `LinearAlgebra` is split into a separate package, then this looong document can be split into multiple pages, which will better organize the document and improve readability.

File: stdlib/LinearAlgebra/docs/src/index.md [hunk 1]
```diff
@@ -527,67 +527,105 @@ the input argument belongs on (`side`). The possibilities are:
...
 | `'U'`       | The diagonal of the matrix `X` is assumed to be all ones. |
 
+
 ```@docs
 LinearAlgebra.BLAS
+LinearAlgebra.BLAS.set_num_threads
+LinearAlgebra.BLAS.get_num_threads
+```
+
+### Level 1 BLAS functions
```
Comment (fredrikekre) [inline stdlib/LinearAlgebra/docs/src/index.md:537]: If you are adding these subheadings I think you should explain what it means.

**Comment by ViralBShah** [comment]
We should split it into multiple pages anyways - can't we already do it now? It will definitely be much easier in a separate repo, where we can move faster. 

@KristofferC what do we need to do to move LinearAlgebra to a new repo? I always forget a few steps.

File: stdlib/LinearAlgebra/docs/src/index.md (hunk 1 above)
Comment (inkydragon) [reply stdlib/LinearAlgebra/docs/src/index.md:541]: OK, I've added some simple explanations and links.

**Comment by inkydragon** [comment]
> can't we already do it now?

After some trying, I found some hard codes.
https://github.com/JuliaLang/julia/blob/fd70eabc1dd6c79149edb82da9e9c37d89e249d9/doc/make.jl#L27-L32

It looks like some modifications are needed to support multiple source files.

**Review by ViralBShah — APPROVED** [review]

**Pull request merged**